pyproject.toml

```diff
@@ -88,10 +88,59 @@ parallel = true
 
 [tool.coverage.report]
 exclude_lines = [
-    "@abc.abstractmethod",
-    "@overload",
-    "if (typing\\.)?TYPE_CHECKING:",
-    "pragma: no cover",
-    "raise NotImplementedError"
+  "@abc.abstractmethod",
+  "@overload",
+  "if (typing\\.)?TYPE_CHECKING:",
+  "pragma: no cover",
+  "raise NotImplementedError"
 ]
 skip_covered = true
+
+[tool.ruff]
+target-version = "py39"
+exclude = ["migrations"]
+lint.select = ["ALL"]
+lint.ignore = [
+  # Safe to ignore
+  "A001", # Variable is shadowing a Python builtin
+  "A002", # Function argument is shadowing a Python builtin
+  "ANN", # Missing Type Annotation
+  "ANN401", # Dynamically typed expressions (typing.Any) are disallowed in `**kwargs`"
+  "ARG001", # Unused function argument (request, ...)
+  "ARG002", # Unused method argument (*args, **kwargs)
+  "ARG005", # Unused lambda argument
+  "D", # Missing or badly formatted docstrings
+  "E501", # Line too long (>88)
+  "EM101", # Exception must not use a string literal, assign to variable first
+  "EM102", # Exception must not use an f-string literal, assign to variable first   "ERA001", # Found commented-out code
+  "ERA001", # Found commented-out code
+  "FBT", # Flake Boolean Trap (don't use arg=True in functions)
+  "N999", # Invalid module name
+  "PLR091", # Too many statements/branches/arguments
+  "PLR2004", # Magic value used in comparison, consider replacing <int> with a constant variable
+  "RUF012", # Mutable class attributes https://github.com/astral-sh/ruff/issues/5243
+  "SIM105", # Use contextlib.suppress(ImportError) instead of try-except-pass
+  "SLF001", # private-member-access
+  "TRY003", # Avoid specifying long messages outside the exception class
+
+  # Should be fixed later
+  "C901", # Too complex
+  "N802", # Function name should be lowercase
+  "N803", # Argument name should be lowercase
+  "N806", # Variable should be lowercase
+  "PERF401", # Use `list.extend` to create a transformed list
+  "S101", # Use of 'assert' detected
+
+  # Required for `ruff format` to work correctly
+  "COM812", # Checks for the absence of trailing commas
```

Comment:
Btw, `ruff format` works fine with this rule, the warning (that can be ignored) is basically just implying that you might need multiple rounds of `format` + `check` with this rule enabled (since a trailing comma, being a "magic comma" implies that args need to be on different lines, etc.

Reply:
Mmh, its spitting out the warning all the time though, which is annoying. I keep it in for now, I like that single line arguments don't have a trailing comma. Good to see here. 

<img width="1090" height="188" alt="CleanShot 2025-07-25 at 11 03 31@2x" src="https://github.com/user-attachments/assets/1641cec3-a8cb-42eb-bfa7-b3ab55fc5b3a" />